Services/data/devicemodels/scripts/SetFuelLevel-method.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+/*global log*/
+/*global updateState*/
+/*global updateProperty*/
+/*global sleep*/
+/*jslint node: true*/
+
+"use strict";
+
+/**
+ * Entry point function called by the simulation engine.
+ *
+ * @param context        The context contains current time, device model and id
+ * @param previousState  The device state since the last iteration
+ * @param previousProperties  The device properties since the last iteration
+ */
+
+/*jslint unparam: true*/
+function main(context, previousState, previousProperties) {
```

Comment:
why do you need prevstate & prevprops?

Reply:
We don't, but the javascript interpreter uses this calling convention (see Invoke() in JavascriptInterpreter.cs). I don't think that it makes sense to update the JS interpreter code to know when to call a function with fewer arguments (it would be hard to identify when the function will or won't use the state arguments), and if I leave these arguments out in the method implementation, then we've got an ambiguous function call from the interpreter, which I'd like to avoid (am I calling main(context, previousState, previousProperties), or am I calling main(context)?).